src/dagmc/DagMC.cpp

```diff
@@ -39,36 +39,32 @@ const bool counting = false; /* controls counts of ray casts and pt_in_vols */
 const std::map<std::string, std::string> DagMC::no_synonyms;
 
 // DagMC Constructor
-DagMC::DagMC(Interface* mb_impl, double overlap_tolerance, double p_numerical_precision) {
+DagMC::DagMC(std::shared_ptr<Interface> mb_impl, double overlap_tolerance, double p_numerical_precision) {
   moab_instance_created = false;
   // if we arent handed a moab instance create one
-  if (NULL == mb_impl) {
-    mb_impl = new moab::Core();
+  if (nullptr == mb_impl) {
+    mb_impl = std::shared_ptr<Interface>(new moab::Core());
     moab_instance_created = true;
   }
 
   // set the internal moab pointer
-  MBI = mb_impl;
+  MBI = std::shared_ptr<Interface>(mb_impl);
 
   // make new GeomTopoTool and GeomQueryTool
-  GTT = new moab::GeomTopoTool(MBI, false);
-  GQT = new moab::GeomQueryTool(GTT, overlap_tolerance, p_numerical_precision);
+  GTT = std::shared_ptr<GeomTopoTool> (new GeomTopoTool(MBI.get(), false));
```

Comment:
Should these be `unique_ptr`'s as those objects are going to carry state that is related directly to calls that have been made on this DagMC object (e.g. `init_obbs`)?